pyNN/neuron/standardmodels/electrodes.py

```diff
@@ -84,13 +84,44 @@ def _update_iclamp(self, iclamp, tstop):
                 else:
                     amp_val = self._h_amplitudes.x[int(ind)-1]
                 self._h_times.insrt(ind, tstop)
-                self._h_amplitudes.insrt(ind, amp_val)                
+                self._h_amplitudes.insrt(ind, amp_val)
 
             self._h_amplitudes.play(iclamp._ref_amp, self._h_times)
 
+    def _round_timestamp(self, value, resolution):
+        return int(value/resolution+0.5) * resolution
+
+    def _check_step_times(self, times, amplitudes, resolution):
```

Comment:
This appears to overlap substantially with the equivalent function in pyNN.nest. Can this redundancy be eliminated?

Reply:
Moving this method from each simulator files to the `StandardCurrentSource` class.